packages/connect-node/src/compression.ts

```diff
@@ -81,12 +81,22 @@ function wrapZLibErrors<T>(
         break;
       case "Z_DATA_ERROR":
       case "ERR_PADDING_2":
-        // TODO(TCN-785) we may want to preserve the cause, but not in the connect error message
-        e = new ConnectError("decompression failed", Code.InvalidArgument);
+        e = new ConnectError(
+          "decompression failed",
+          Code.InvalidArgument,
+          undefined,
+          undefined,
+          code
```

Comment:
We want to maintain a chain of errors, and have to pass the unmodified error as the cause here.

This makes the following piece of code continue to work, whether we wrapped the error or not:

```ts
const isNodeEOF = unwrapNodeErrorChain(err)
  .map(getNodeErrorProps)
  .some((p) => p.code == "ERR_STREAM_WRITE_AFTER_END")
```